### PR TITLE
Resource Oriented AI Proposal - AI That Cares About The Station

### PR DESCRIPTION
**Pull request by SaphireLattice**
There we go, Yet Another AI Proposal. This one based on one of the existing ones, at least in spirit. 

Goal is to make AI have... a bit more to chew on and care about. It shouldn't be utterly detached from the station, it really should care about the state of it, and that it's powered, not on fire and there's people on it. The proposal should encourage interactions with other departments, and just people in general. Also discourage AI players from being just an extra security officer with ghost flying.

Not sure if I've accomplished all of that here, but here's hoping for at least a step into a good direction.

TODO:
- Intellicards, and overall "control points" for AI
- *[Reader-Write-In]*

### CODE REVIEW

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+- Left-click or alt-click on vents to unlock them, for cheap
+- **TODO**. Ideas: sensor sweep? Mass applying (un)bolting to everything an APC’s connected to? Atmospherics reset action that can be used from the action bar/alt-click radial menu and applied to an atmos alarm to reset it to default? Clicking vents to unlock them?
+- Consoles! They aren’t free and cut a decent chunk, to encourage the AI player to choose a Compute option, but in a pinch this should work too. It also gets paid by the player having to manually locate a console and being unable to check on things elsewhere but the vicinity of the console.
+
+### Sources
+
+- AI specialized machinery, contributing a significant amount of processing capability to the AI.
+  - The downside to this is that it requires someone physical to build it, and needs a decent amount of resources
+  - Such machinery should be difficult to destroy or even move. It stays where it stays, unless the situation requires it and someone puts enough effort to move it; or break it.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:111]: It can easily just weigh two metric tons, same as AI core, to avoid being able to move it easily. You should be able to move it to rearrange a room but moving it across the station should be painful. I believe the solution to the AI dragging their core into maints and hiding it (to activate the doomsday device willy nilly) was the pinpointer pointing to them?

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  - Generates heat up to a throttle cutoff, and works better when cooled. When not in an optimal environment, provides less computing power, but still comparable to servers.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:112]: Agree. Gives Atmos (or just the AI and smart synced angiborgs something to do) and *cough cough* ties into my HVAC proposal (#319).

In discord I touched on the fact that radiators designed for convection are *not the same* compared to radiators designed for radiation. In reality, computers in space would insulate themselves, so you could make spaced servers overheat themselves.

Comment (ArtisticRoomba) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:112]: PS: This also gives Science something to do (as one of the options you have to help the crew is just glorified bitcoin mining)

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  - Backup power? Camera built in?
+- Crew devices; computers, machines, servers
+  - These don’t contribute as much **Compute** as they are not specialized for AI use. Servers do provide more **Compute** than just consoles, and provide **Bandwidth** in amounts comparable or identical to dedicated machines..
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:115]: Very good idea.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  - Locks out the device for crew use, rendering it inoperable. This acts as a conflict point with the crew. Computers don’t work, machines either. Servers don’t provide normal functionality or do so in a very limited way.
+    - E.g. crew monitoring server could give invalid/partial data with a warning, or just not work at all. Same for telecom servers.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:117]: I'd like to think of the Malf AI slowly infecting machines as like a virus. The little APCs and pointless vending machines hacked don't provide much compute or bits by themselves, but all together, it turns into a *very big* problem for the crew, and a sweet benefit to Malf's compute and bits.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  - Non-antag AI can be flushed out from the device via a near-instant crew exposed action. This includes a subverted AI. Could be an UI that shows “This device is in use by AI \<name\>. \<Override (button)\>.”
+  - Such shooing away should keep the AI from taking the device over for at least some time, at least a minute. It should not be able to just annoy the crew to death even when compromised - it’s still shackled.
+  - Antagonist AI, “Malf”, should have an option or be able to spend some resources to completely lock the crew out. Will require more detailed balancing later.
+  - For “Malf”, the timeout should be reduced or bypassable. Again, might need some balance tweaking whether it’s just a pop up “you sure?” or a resource sink.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:121]: Good point. Is Malf AI taking over pointless, little machines, one by one, a source of resources, or will it require Bits/s? IMO it should the latter, as this might provide a way for Malf AI to claw some much-needed processing power in its final moments (when it's very obvious that it's Malf). The crew could just shoot the vending machines or snip the AI wire to cut out the gained processing power. Could be something cool to see engineering do in an effort to combat an active Malf.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+- Borgs! Useful and somewhat plentiful, perfect for an emergency ~~snack~~ method of survival when the station’s powered off and the AI SMES is about to drain, with a downside of rendering the borg unable to synthesize voice (speak) when AI’s along for the ride, except for beeps and emotes. Does not require borg sync, but requires approval from the borg even if synced. Malf AI can bypass the approval.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:122]: This needs to be fleshed out/explained further cause I don't know the details of this, it's hard to follow.

So an AI can hop in a borg's body with consent from the Borg player (or just automatically if there's a Posi brain waiting for a ghost), but the AI can't speak, only beep/emote. And it doesn't require sync.

Should this control require an obscene bits/s? This is giving AI a pretty direct way of control over the physical world.

When a Malf AI hacks a borg, does that force a sync? Does it appear synced to crew, or will it cause a fake sync?
For the purposes of metagaming, can a borg choose to sync/unsync itself at will, or will it require crew intervention? 

Probably the latter, as it makes it good for lowpop and metagaming. A borg should require crew intervention to unsync if synced to a Malf AI, but otherwise, it can unsync itself at will (if it would ever want to).

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+The AI satellite has several of the AI machines built into it, providing the base necessary resources for the AI, and some extra to make it actually useful to the crew.
+
+**Note:** Something noted by other people is how this system would interact with multiple AI. Two AI might be possible with either significant crowding and cut down resources, or a very annoyed crew as two AI fight one another for scraps. They should be able to lock one another out from a device, and require significant investment of power to pull control away. Of course, the crew can always just shoo them away with same near-instant action.
+
+## Vision - Cameras and Map
+
+The AI is not a biological entity, and it’s only barely a “physical” one. It doesn’t need to care about specifics of how things look, but rather whether or not an area is intact, safe, and secure. As such, the vision system is an optional ability that an AI player can shut down, whether to free space up for something else, or as an emergency measure.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:130]: Agree. This requires the NavMap to be fleshed out further (of course), which you describe below.

AI should be able to do a LOT, even without TV vision.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+### Map Vision
+
+The “map” view replaces the entirety of the AI player's vision if the camera ability is shut down, or if there’s no cameras *and* they activate a composite vision ability. It should provide information about the station’s hull - walls, floors (or a glaring lack of one or being stripped down to a lattice), windows. It also should display machines that have an AI control wire - deemphasizing them if the wire’s been snipped, and potentially not showing them at all if there’s no camera that can see the machine (to make camera snipping actually work around APCs, air alarms). Fire alarms and atmosphere alarm status should potentially be shown over the area too - perhaps as an ability.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:134]: > and potentially not showing them at all if there’s no camera that can see the machine (to make camera snipping actually work around APCs, air alarms)

Disagree. AI should have control over everything on the station if there's an AI wire, it shouldn't *need* to see a TV view of the device it wants to control. It should have an intrinsic "feel" of the device (because it's wired through an AIC wire directly to it). If crew wants to disable AI control, they have to enlist engineering (or the tiders) to start running around and snipping wires.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+While using the AI map vision, the crew can be seen via suit sensors, only if they have tracking on. Otherwise AI is unable to tell if someone is in the area without using cameras. A potential ability idea is to give cameras a motion detection cone that gets rendered for AI, and highlight entities within it - differently from tracked crew that enters it.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:136]: Would be good and tie into motion alarm alerts. For stuff like feral slimes, spiders, crew with a cryptographic sequencer in Captain's Quarters, and crew in blood-red hardsuits holding C-20Rs in the Armory.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+The map vision should not use “#00FF00 on #000000” color scheme for purpose of reducing eye fatigue. Consider dark dark gray background as a default for areas that are not space exposed. Preferably implement a way to set custom colors for your UI.
+
+### Cameras
+
+Cameras cover almost all of the crew areas, places where most people work and exist within, to monitor their wellbeing and the state of the station. They should not be nearly free to build, and should require an autolathe producible board, some steel, LV wire.
+
+They should not be more durable than a person - a hard enough knock should deactivate it for a small period of time, while sustaining enough damage will keep it offline until repaired. A non-violent option for keeping cameras offline should be available, using cloth and/or gauze makes it impossible to see through a camera, taking a relatively long time to cover it completely (3 or 5 second doafter, balanced by it being a “stealth” option). A camera disabled in such a way should still be shown as active for AI and warden. Though it should have no motion detection cone and convey such explicitly.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:144]: Should there be reinforced cameras or emplaced wall cameras for things like Armory/Perma/AI Core? They would have a reduced vision code (as they wouldn't be mounted one tile out from a wall and wouldn't be able to see behind them/in a wider cone as such). These would be much harder to break, and might be more difficult to cover up.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+Power is required for cameras to work at full capacity, reducing their vision to half or third of their normal range when it is not available. Though power being out is a bigger issue for AI itself than just the cameras.
+
+## Borgs - Sync and control
+
+The AI has no way to affect anything physical (at least, a shackled one - not malf). Its primary way to do anything in that regard is borgs, and directing the crew. While the crew has own concerns, like air, health and such, borgs are hardier and are almost always available for AI to use.
+
+Borgs can opt into synchronizing their laws with AI, gaining map or camera view from AI (todo: detail the incentives). This does not make them less susceptible to ion storm law corruption, but the AI can fix the laws by rewriting them to its own at some cost.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:152]: Agree. Borgs should also get something similar to remote AI control (having the ability to remote interact with things in their viewport, as well as having camera view from AI). This means that borgs could see into other rooms, remote open/close doors, and view APCs remotely. Of course, this relies on them being synced and the AI wire uncut.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+Emagging should forcefully but silently unlink a borg from the AI if it’s synced. If it’s not, AI is not relevant.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:154]: This needs to be delved into a lot because there's edge cases for this. This also kinda conflicts with how we want emagged borgs to be found out: through roleplay,

If a borg is emagged, it should break the sync but still fake the sync for AI and crew, basically continuing a one-way sync. The borg gets its benefits it would normally get from being synced (remote control and camera vision). But do we want synced borgs to also have a camera the AI can view? Because if we want the camera vision to cut upon being emagged, it would be a bit obvious if a borg is emagged.

Alternatively we could keep borg vision for emagged borgs and have them attempt to fool the AI as much as possible while still following directives.

A person who attempts to EMAG a borg that is synced/hacked by a Malf AI should just fake the EMAG working, otherwise any poor thief/traitor could just "malfcheck" AI. It would also be interesting to not fake the interaction and instead have the EMAG fry, that way roleplay could occur: the AI and thief/syndie would have to barter with the AI to achieve its goals, otherwise they would rat. But that's in a perfect world of HRP.

Also PS: Make camera vision through borgs take bits/s. Have research for "runtime optimizations" which reduces the bits/s cost for actions. Could also make alternative research for Malf AI with further improvements so Malf AI has more effective power from the same amount of servers. But you could have Malf AI do something "risky" as a way to earn it.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+Crew can cut AI control wire (multiple required?) in borg frames to prevent them from being able to sync to the AI, and letting the crew modify their laws once again (or just resetting to what they had before AI overriding them - including previously present ion laws, if there’s no way to set laws of a borg).
+
+## Assorted ideas
+
+- Paperwork management UI that AI gets *instead* of being able to interact with fax machines. It would list all papers received and let the player read them. It would also allow editing (if not stamped). Allows sending the same way a fax can.
+- AI stamp that it can put on any received paper through paperwork management UI. If it fits into the art design, make it a QR or DataMatrix code that just says “AI”. Otherwise just have it be a text.
+- Holopads are a way to communicate with the crew for free, with full vision and such. The downside is that it completely takes over the AI’s viewpoint for the duration of a personal call. Just formalizing the existing implementation by Chromiumboy here.
```
Comment (ArtisticRoomba) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:162]: All points here are cool and I agree.

**Review by ArtisticRoomba — COMMENTED** [review]
Overall I love the proposal, of course as a draft it needs further writing and discussion, detailing the nuances of AI/Malf AI, how it would interact and benefit the crew, what it would look like in a round, etc. Sweet design doc, I highly look forward to gameplay like this.

File: src/en/space-station-14/round-flow/proposals/station-ai.md (hunk 6 above)
Comment (SaphireLattice) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:122]: This is not a way of *controlling* a borg's body at all, this is specifically to use them as Compute generating entity. I don't have any plans for letting AI just have a physical body, this defeats entire point of this proposal. Any physical body providing thing would basically cut the AI off its network and give you effectively yet another borg.

Good points on borg sync though.

Comment (ArtisticRoomba) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:122]: Ahhhhhh, I see now, that makes much more sense.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+
+Meanwhile in SS13, the AI as a concept also integrates rather heavily with an antagonist role “Malf”, a flip side to having an AI on the station. Or rather it does *not* integrate. The antag is not designed well in SS13, and is metagamed to hell. There’s no way to metashield it because you would need to meta shield pretty much *every single thing* that “Malf AI” can do. It’s also susceptible to just being rushed by power-gamers the moment it’s detected for sure. It’s a stationary target in a game where any attempt at solo “tower defense” is bypassed with 2 C4 charges, barring admin intervention.
+
+This proposal is based in part on [Moony's AI proposal](https://github.com/space-wizards/docs/pull/160).
+
+## General overview
+
+The AI is a big eye in the sky capable of quickly observing any point on the station, and acting on it. It’s shackled by the silicon laws, yet can be subverted or be a round start antagonist. It’s a fully digital entity - a standard AI can not affect the physical world outside of a few specialized devices for such.
```
[the remark this answers is not in the document]
Comment (hyphenationc) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:22]: should be a posibrain imo, intellicard is a temporary storage device. otherwise agreed, needs a death gradient

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+- Radio system (listening) - Binary is free! But the rest AI has to buy, requiring a tiny bit of resources for each channel. Comes as part of the spawn-in preset for standard AA radio.
+- Radio system (speaking) - Binary is, again, free! This is distinct from listening as to not cripple the AI player entirely if they have to discard radio, but still force to sacrifice something to gain resources. Part of spawn-in preset with AA.
+- Vision system - Without it, there’s just the “Nav UI”-like map replacing camera view. Can still see computing systems, doors, etc that have an AI wire. Maybe see Air Alarms? Spawn preset bundled. (MAYBE). See the [Vision](#vision---cameras-and-map) section for more details.
+- Vision system composition - Map UI can be an extremely powerful tool in gameplay perspective. As such it should be an extra ability to composite it over where the cameras are missing. Balancing note - shouldn’t be bypassable by player toggling camera vision rapidly on/off.
```
[the remark this answers is not in the document]
Comment (hyphenationc) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:90]: ...adding some AI capabilities behind researches is also a REALLY good idea tbh...

**Comment by JRPirks** [comment]
Code for methamphetamines
Code for beers

**Comment by hyphenationc** [comment]
> Code for methamphetamines Code for beers

wha?

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+
+As described above, AI is a fully digital entity. It needs processing resources to continue to exist, especially in its full form. Without these it will be crippled or straight up start dying.
+
+The computing resource system consists of two resources: Compute, and Bandwidth.
+
+**Compute** is the constant upkeep resource, something that the AI requires by itself to exist, and is used for abilities that stay active indefinitely, until toggled off. It is a pool that gets its capability expanded by having more machines under AI’s control, and gets consumed by abilities until they are turned off.
+
+**Bandwidth** is the momentary resource. It’s used for having something happen right now, near-instantly, and something that does not last for long. It’s a pool with a maximum size extensible by controlled machines, and every momentary ability consumes some of the resource from it. It regenerates at a base rate, further enhanced by the amount of unallocated **Compute**.
+
+*Alternative*: Shared pool that gets extended/shrunk based on controlled devices, and chunks of it are allocated persistently while unallocated part can be used for temporary abilities. Something to playtest later!
```
[the remark this answers is not in the document]
Comment (frobnic8) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:61]: You might call this "memory" as opposed to "bits".

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+- Being unable to observe anything is a much bigger issue. This drops AI down to a schematic view with maybe some critical infrastructure being visible to it in a vague way. It can still claw its way back, but it will not be easy. And again, this renders it effectively useless without help.
+- Being unable to exist is self-explanatory.
+
+The resource system should encourage an AI to either request extra processing machines for itself, or to start using the station infrastructure to power its abilities. Or as a last resort, start to micromanage and drop active abilities. As such the prices of things should be balanced accordingly. An AI that wants to do more than just listen, talk and speak shouldn't be able to just stay only with round start resources.
+
+A possible way to accomplish that is to make vision and speech capabilities rather huge in resource demand, while the core AI footprint is relatively small and could fit on about 4-8 consoles or APCs, or 1-2 borgs, without losing integrity.
```
Comment (frobnic8) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:79]: I love the idea of the AI having the OPTION to temporarily turn off vision or speech to have enough capacity to do something normally out of it's limits, then hope things are fine until they can turn them back on.

**Comment by walksanatora** [comment]
people really should link the documents so people dont have to go searching.
[Link to rendered](https://github.com/SaphireLattice/Fork.SS14.Documentation/blob/proposal/ai/src/en/space-station-14/round-flow/proposals/station-ai.md)

**Comment by walksanatora** [comment]
> not integrated with it as borgs have no requirement to obey or even listen to AI

this is one thing starlight fixxes. it adds a law 0 to borgs. "follow Station AI orders" meaning that station AI is more important then crew saftey. normally the laws allign. but antimov changes all borgss into killing machines.

**Comment by walksanatora** [comment]
so right it is intended that if you exert no more resources. every station's AI should be able to
1. speak.
2. observe.
3. exists.
without outside interferrence (aside from obvs someone keeping power on)

**Comment by hyphenationc** [comment]
@walksanatora if you're gonna do a review of this, could you please use the grouped review thingy (go to Files Changed, find the first line you wanna start changing and click the + on its left side, and make sure to click Start a Review)? This'll avoid excessive emails to subscribed individuals, and also avoid flooding the github channel for the wizden discord. Thanks, mate.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+
+Being a digital entity the AI requires processing resources to sustain its base capabilities and any extra it might want. This resource management would be centered around machines that provide them - either specialized ones that do so at maximum efficiency; or using crew oriented devices, temporarily locking the crew out in the process.
+
+The antagonist, “Malf” AI, must not be immediately distinguishable from a normal AI. It has access to abilities that affect the physical world way more, and that push the resource management element even further. Is AI asking for more servers because it wants to help the crew even more, or is it because it plans to detonate the nuke?
+
+Multiple AIs are not in the design scope of this document. They should however be possible, and all of the systems should NOT treat AI as a singleton. Machine takeover system could be reused, for example to implement minor “virus” ghost role antag/NPC.
```
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:30]: so then it becomes a war of AI versus AI. and if you could eg: make a AI server and upload a AI to it but then cut the AIA wires to "lock" the other Malf AI out and lock the crew alligned AI in it would make it so then the Malf AI would need to force a borg to get over there and physically remove the Crew-alligned AI.

File: src/en/space-station-14/round-flow/proposals/station-ai.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:61]: yeah I think Memory is a good name for it. and I love the idea of in the final moments the AI player having to ration out their resources very carefully to try and save themselves.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+
+### Usage
+
+Nothing is a free action.
+
+Primary use of the computing resources is ***keeping the AI alive***. It lives or dies through the lack of the processing power. The secondary use is actually letting it do things on the station. Having no processing power should hurt, but not an instant death.
+
+The definition of “alive” here is fuzzy. The AI exists to serve (or kill, for antag) the crew. For that it needs to be able to act, observe, and exist.
+
+- Being unable to act is the softest “kill”. And “speaking” is not a free action, though pretty close to it. It still can observe and maybe scrape resources together to shout out a warning. But it is effectively useless without someone else helping it.
```
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:73]: which the basic operations are the current station AI features right.
1. observation
2. all-access radio
3. bolt/shock/emergency some doors.

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+- Being unable to observe anything is a much bigger issue. This drops AI down to a schematic view with maybe some critical infrastructure being visible to it in a vague way. It can still claw its way back, but it will not be easy. And again, this renders it effectively useless without help.
+- Being unable to exist is self-explanatory.
+
+The resource system should encourage an AI to either request extra processing machines for itself, or to start using the station infrastructure to power its abilities. Or as a last resort, start to micromanage and drop active abilities. As such the prices of things should be balanced accordingly. An AI that wants to do more than just listen, talk and speak shouldn't be able to just stay only with round start resources.
```
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:77]: i feel like open/closing doors sshould be a low-cost action. with actions like emergency access and bolting being real cost.

File: src/en/space-station-14/round-flow/proposals/station-ai.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:90]: yeah and a overlay for power and disposal networks for when places get blown up. again engineering would sing your praises. and also as MALF that would be something you could use as bait for more resources. "you want me to have better ways to diagnose issues", "give more computing bandwidth and I will work on it"

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+- Radio system (listening) - Binary is free! But the rest AI has to buy, requiring a tiny bit of resources for each channel. Comes as part of the spawn-in preset for standard AA radio.
+- Radio system (speaking) - Binary is, again, free! This is distinct from listening as to not cripple the AI player entirely if they have to discard radio, but still force to sacrifice something to gain resources. Part of spawn-in preset with AA.
+- Vision system - Without it, there’s just the “Nav UI”-like map replacing camera view. Can still see computing systems, doors, etc that have an AI wire. Maybe see Air Alarms? Spawn preset bundled. (MAYBE). See the [Vision](#vision---cameras-and-map) section for more details.
+- Vision system composition - Map UI can be an extremely powerful tool in gameplay perspective. As such it should be an extra ability to composite it over where the cameras are missing. Balancing note - shouldn’t be bypassable by player toggling camera vision rapidly on/off.
+- Comms console - Separate and cheaper than others as it’s rather important and relatively low impact (except round stalling, oops?). Though would it be better to keep it MORE expensive?
```
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:91]: kind of like nuclear war operatives. "we know the station is under-threat, stay there and resolve it. we will come pick you up when it is over"

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+- Assorted consoles - Just a general category but they each should not be free.
+- Bolting a door - it’s an override in case of an emergency, not a go-to that can be spammed with impunity. Bolting off giant hallways would require managing resources, and making sure to unbolt things elsewhere. Keeping things bolted is possible, but not unless you the player wants to do nothing else.
+- Research - a resource sink that takes a lot of processing capability of the AI in both of the available resources at a base, and can be increased further (powers of two? exponential?) to provide more research. Should not be a way to completely ignore science department gameplay, but rather be a side option, requiring substantial resource investment to provide good enough amount of points.
+- Cargo bounty - Call it “Bluespace Calculation” or any other fancy name. Basically the same as research but also gives out something that can fulfil a cargo bounty. Giving it to cargo could be done through integration with the [paperwork management UI](#assorted-ideas)? Should be separate from research resource sink to not let AI just double-dip.
```
[the remark this answers is not in the document]
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:95]: I think the idea would be that cargo would fax a cargo bounty to the AI. and the AI can go "let me work on it"
and attempt to basically hack a item from somewhere to cargo by performing complex bluespace math. "who cares if it is stealing, we just got paid, and it isn't our stuff being stolen". this would also add a REALLY funny random event. where random items can just dissapear because some other station's AI just Bluespace Stole stuff for a cargo bounty.

File: src/en/space-station-14/round-flow/proposals/station-ai.md (hunk 3 above)
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:115]: thatway a good (Malf) AI could in theory hide in the misc devices of the station. crying to the void asking for repairs and upgrades.

File: src/en/space-station-14/round-flow/proposals/station-ai.md (hunk 12 above)
Comment (walksanatora) [reply src/en/space-station-14/round-flow/proposals/station-ai.md:154]: i would love the faked borg basically having to pretend to be a Emagged borg. but yet still being MALF alligned. it now has to balance 2 fake allegiances (the crew, and the synicate) and it's true allegiance (the MALF ai). mabey should have some way of storing who attempted to emag them in a law. so it is easier to remember (there have been many times as a EMAG'd borg where I forgot who owns me and so I have to check law 0)

File: src/en/space-station-14/round-flow/proposals/station-ai.md [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+
+Borgs can opt into synchronizing their laws with AI, gaining map or camera view from AI (todo: detail the incentives). This does not make them less susceptible to ion storm law corruption, but the AI can fix the laws by rewriting them to its own at some cost.
+
+Emagging should forcefully but silently unlink a borg from the AI if it’s synced. If it’s not, AI is not relevant.
+
+Crew can cut AI control wire (multiple required?) in borg frames to prevent them from being able to sync to the AI, and letting the crew modify their laws once again (or just resetting to what they had before AI overriding them - including previously present ion laws, if there’s no way to set laws of a borg).
+
+## Assorted ideas
+
+- Paperwork management UI that AI gets *instead* of being able to interact with fax machines. It would list all papers received and let the player read them. It would also allow editing (if not stamped). Allows sending the same way a fax can.
```
Comment (walksanatora) [inline src/en/space-station-14/round-flow/proposals/station-ai.md:160]: I like the AI able to do paperwork. I could see the RP of the station AI being assigned as a beurocrat in charge of keeping documents in check. and mabey MALF AI could get a "cybersun pen" upgrade that allows them to edit stamped documents like syndicates. I could imagine how chaotic that would be. even more so if the AI could eg: intercept all Faxes to CentComm and start acting as CentComm and giving superior orders. but that would only reallly play out on MRP/HRP servers. cause who uses faxes in LRP.
